### ISSUE TITLE
Should assigning an unawaited future to `let _ = ...` be it's own lint?

### ISSUE THREAD

**Issue by SquareMan** [rust-lang/rust-clippy#9721]
### What it does

I recently came across and fixed a bug in my code where I had an async function that returns a `Result` and meant to ignore the result so I used `let _ = ...`, but forgot to `.await` the function. ([link to full context](https://github.com/BfBB-Clash/BfBB-Clash/pull/107/files).

I know that lints like `let_underscore_drop` exist but this is quite broad and doesn't do what I would want here.

Should there be a new lint that specifically looks for assigning a `Future` to `let _ = ...`?

### Lint Name

let_underscore_future

### Category

suspicious

### Advantage

- Alerts the user that a created `Future` is not being completed.

### Drawbacks

Is there any reason why you might want to immediately drop an un-awaited future with `let _ = ...`? I'm not aware of any.

### Example

```rust
async fn foo() -> Result<(), Error> {}

async fn main() {
  let _ = foo();
}
```

Could be written as:

```rust
async fn foo() -> Result<(), Error> {}

async fn main() {
  let _ = foo().await;
}
```

**Comment by agostbiro** [comment]
Hi, thanks for this lint!

I just came across a case where I think it produces a false positive:

```rust
#[tokio::main]
async fn main() {
    let _ = tokio::spawn(async {
        println!("spawned")
    });
}
```
[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=d5ee8638bb9c619f1c82aa5308faddde)

In this case the returned [JoinHandle](https://docs.rs/tokio/latest/tokio/task/struct.JoinHandle.html) is a Future, but it [doesn't have to be awaited](https://docs.rs/tokio/latest/tokio/fn.spawn.html) to make the provided future start execution.
